refactor(payments): Convert SubscriptionTitle component to utilize Tailwind

xlisachan opened the pull request:
## Because

- we are converting to Tailwind

## This pull request

- [x] Replaces all CSS/SCSS with Tailwind utility classes
- [x] Removes .css and .scss files
- [x] Reworks Storybook files
- [x] Updates responsive design 

Additional Components
- `PaymentErrorView` - removed margin-top
- `PaymentProcessing` - removed unnecessary classes (`container`, `card`), added margin
- `NewUserForm` - color-corrected shield icon
- `PlanDetails` - color-corrected box-shadow (NOTE: update responsive ticket to add potential to DRY up box-shadow), revised `coupon-info` to create `green-icon-text` which is now used in `SubscriptionTitle` as well
- `SubscriptionCreate` - color-corrected box-shadow (NOTE: need to create ticket for this component)
- `mixin` - color-corrected box-shadow (NOTE: update ticket to revisit  stylesheets with `subscription-title`)
- payments' tailwind stylesheet - created `green-icon-text` to use in both `PlanDetails` and `SubscriptionTitle` 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.

## Screenshots (Optional)
SubscriptionTitle
<img width="916" alt="Screen Shot 2022-09-01 at 12 11 44 PM" src="https://user-images.githubusercontent.com/28129806/187962179-3243ce02-0855-4b6b-81c3-51e23f7924ed.png">

Checkout
- Mobile
<img width="943" alt="Screen Shot 2022-09-01 at 12 13 06 PM" src="https://user-images.githubusercontent.com/28129806/187962454-de0e7cdc-cc0c-4763-afe9-131a4d4e6b9a.png">

- Desktop
<img width="2124" alt="Screen Shot 2022-09-01 at 12 13 47 PM" src="https://user-images.githubusercontent.com/28129806/187962598-c561b432-6b84-4c1e-b953-88eec9bb48c7.png">

PaymentConfirmation
- Mobile
<img width="916" alt="Screen Shot 2022-09-01 at 12 14 44 PM" src="https://user-images.githubusercontent.com/28129806/187962767-6e4fb491-60d6-4ed6-b08a-39f311878f54.png">

- Desktop
<img width="2114" alt="Screen Shot 2022-09-01 at 12 15 20 PM" src="https://user-images.githubusercontent.com/28129806/187962890-cc9aacca-d27f-4fe4-99e8-871a5c37e4ac.png">

PaymentErrorView
- Mobile
<img width="944" alt="Screen Shot 2022-09-01 at 12 16 06 PM" src="https://user-images.githubusercontent.com/28129806/187963065-e9c2870e-7f3b-44c2-8c37-2e6939f4a213.png">

- Desktop
<img width="2190" alt="Screen Shot 2022-09-01 at 12 16 48 PM" src="https://user-images.githubusercontent.com/28129806/187963178-fb1a9e66-de2a-44c7-97c8-87b4dce55e7f.png">

PaymentProcessing
- Mobile 
<img width="952" alt="Screen Shot 2022-09-01 at 12 17 49 PM" src="https://user-images.githubusercontent.com/28129806/187963369-ce17c6bf-ede8-4043-98c7-26dd885b94b6.png">

- Desktop
<img width="2195" alt="Screen Shot 2022-09-01 at 12 17 19 PM" src="https://user-images.githubusercontent.com/28129806/187963279-d0259d85-1ff0-4caf-bb2e-4121df350b60.png">

SubscriptionCreate
- Mobile
<img width="935" alt="Screen Shot 2022-09-01 at 12 19 41 PM" src="https://user-images.githubusercontent.com/28129806/187963711-c9a58c7a-f504-4b17-84ab-9832af5b8f96.png">

- Desktop
<img width="2163" alt="Screen Shot 2022-09-01 at 12 19 08 PM" src="https://user-images.githubusercontent.com/28129806/187963607-385af38a-56d9-47b8-8565-38d3d7f52847.png">

sardesam commented on packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss:
```diff
@@ -6,32 +6,32 @@
   }
 }
 
 .product-payment {
```
I can't select some of the lines so I'll address all suggested changes to reduce DRY in this selector here. Bear with me and let me know if you have any questions:
1. Remove both media queries from **L45-L55**.
2. Move **L33-L37** below **L42**.
3. Remove **L39** and **L43** (now **L48** after adding the lines from (2), I believe).
4. Add two declarations below **L32**: `min-height: auto;` and `margin: auto;`.
5. Wrap **L32** and the two new declarations from (4) in a `@include min-width('tablet')`

sardesam reviewed (changes requested):
Excellent work! Changes look good and nothing else is needed specifically for the `SubscriptionTitle` component.

After careful review of all the changes and testing styles, functionality, responsiveness (including, to the best of my ability, cross browser and device), and as per our Slack conversations this past week, I am completely in line with your suggested next steps; please go ahead and file those tickets if you haven’t already done so.

I did find some improvements to reduce DRY that will lessen the work load later on. I have added individual comments. Let me know if anything needs to be clarified or if the suggested change(s) might have an undesired effect elsewhere. Alternatively, if you think the change(s) are better to address in another ticket, tag here for reference.

One thing I did notice, and I am not sure if it requires updates to `SubscriptionTitle`, `PlanDetails`, etc. or both, is the box shadow on individual containers when trying to make `SubscriptionTitle` appear flush with the content below. It either creates a faint grey line or, if you make the box shadow white, the side borders show a slight gap and the bottom box shadow looks cut off. I wasn’t sure where the culprit was when playing around with it. Let me know if you have any ideas!

Nit: after making some of the revisions, there will be extra spaces in the file. They can be removed 😜 .

N.B. Would advise starting from bottom of each file for line numbers to make sense. I tried to keep it in line with what's on git but unless the lines had been updated, it won't let me select them.

xlisachan commented on packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx:
```diff
@@ -23,22 +22,27 @@ export const SubscriptionTitle = ({
   return (
     <div
-      className={`subscription-title ${className}`}
+      className={`subscription-title bg-white shadow-sm shadow-grey-300 text-center mt-0 mx-4 mb-auto pt-5 pb-px border-y-auto z-1 row-start-1 row-end-2 tablet:mx-0 ${className}`}
```
Notes have been added to checkout-title and checkout-view tickets.

xlisachan commented:
Thanks for the feedback, @sardesam! Aligned with your changes - I also made the following updates:
- Updated `plan-details-component` in the TW file to reflect the border-radius and took them out of the stylesheet
- Removed that gap between SubscriptionTitle/product-payment by changing the margin to 0 0 32px
- Added `px-4` in SubscriptionTitle (SubscriptionUpgradeRoadblock has a really long SubscriptionTitle)
- Updated Storybook (oops) and also included all the variations of SubscriptionTitle

sardesam commented:
> Thanks for the feedback, @sardesam! Aligned with your changes - I also made the following updates:
> 
>     * Updated `plan-details-component-card` in the TW file to reflect the border-radius and took them out of the stylesheet
> 
>     * Removed that gap between SubscriptionTitle/product-payment by changing the margin to 0 0 32px
> 
>     * Added `px-4` in SubscriptionTitle (SubscriptionUpgradeRoadblock has a really long SubscriptionTitle)
> 
>     * Updated Storybook (oops) and also included all the variations of SubscriptionTitle

Fantabulous! Love it! 👍

xlisachan commented:
@sardesam , I made some additional revisions, please see below:
- `plan-details-component` in PlanDetails had properties that did not have any effect on the element (display, flex-direction, grid-row)
- also within `plan-details-component`, `margin` was defined in both the component (which had no effect) and stylesheet, after figuring out the commonalities, I added `plan-details-component` within `styles/tailwind.css` to use in both PlanDetails and PlanUpgradeDetails (see Before/After screenshot below to see margin correction)
- made the box-shadow consistent throughout (`rgb(158, 158, 158)` vs `rgba(158, 158, 158, 1)`)

The rest of the changes were from auto-formatting.

<img width="1005" alt="Screen Shot 2022-09-21 at 10 32 34 AM" src="https://user-images.githubusercontent.com/28129806/191532875-67416979-22cf-4035-8504-2b85f60a3555.png">

sardesam commented:
r+wc

LGTM! 👍 Awesome work! Two nits but otherwise everything checks out: [1]

1. On the checkout page (both new and old subscription flows - i.e. already signed in vs. with `NewUserEmailForm` included), in desktop view, plan details should not have rounded corners on top, to match with subscription setup container on left.

![top-rounded-ss](https://user-images.githubusercontent.com/104124653/191826707-553ea7ea-3188-4b6e-a32b-3363650e0571.png)

2. On the success and payment error pages, in mobile view, plan details should have rounded corners on top, to match the styles for stacked containers (minus the first one that is flush with the top nav).

![no-rounded-ss](https://user-images.githubusercontent.com/104124653/191826711-d100475b-74cf-481c-8cdf-a0a342f2e39c.png)

[1] Note that on the checkout page, in mobile view, plan details becomes flush with SubscriptionTitle, so (2) should only apply to the success and payment error pages.

xlisachan commented:
Thanks for the feedback, @sardesam !

I'll update the border-radius in PlanDetails since I made those changes in this PR thinking the tops should've been rounded in desktop view.

Regarding the border-radius in the Success and Payment Error pages, however, I'll update the tickets that we have for checkout-title and checkout-view, as those changes are within those styles.

sardesam reviewed (approved):
LGTM! 👍
